### DISCUSSION TITLE
[Application Insights] Endpoint w Azure China

### DISCUSSION THREAD

**Discussion by wi5nia** [AzureCommunityPL/group#16] [Ogólna]
Moze case dosc malo popularny wsrod naszego community, ale ja mam okazje ostatnio bawic sie Azure China no i jest... ciekawie ;) ale do rzeczy. 

Jesli wrzucacie skrypt js do strony to trzeba zmienic domyslny endpoint URL dla Application Insights bo w Azure China jest inny. Po moich uwagach support poprawil juz dokumentacje bo oczywiscie "zapomnieli napisac" 😉 

```javascript
<script type="text/javascript">
    var sdkInstance="appInsightsSDK";window[sdkInstance]="appInsights";var aiName=window[sdkInstance],aisdk=window[aiName]||function(e){function n(e){t[e]=function(){var n=arguments;t.queue.push(function(){t[e].apply(t,n)})}}var t={config:e};t.initialize=!0;var i=document,a=window;setTimeout(function(){var n=i.createElement("script");n.src=e.url||"https://az416426.vo.msecnd.net/scripts/b/ai.2.min.js",i.getElementsByTagName("script")[0].parentNode.appendChild(n)});try{t.cookie=i.cookie}catch(e){}t.queue=[],t.version=2;for(var r=["Event","PageView","Exception","Trace","DependencyData","Metric","PageViewPerformance"];r.length;)n("track"+r.pop());n("startTrackPage"),n("stopTrackPage");var s="Track"+r[0];if(n("start"+s),n("stop"+s),n("setAuthenticatedUserContext"),n("clearAuthenticatedUserContext"),n("flush"),!(!0===e.disableExceptionTracking||e.extensionConfig&&e.extensionConfig.ApplicationInsightsAnalytics&&!0===e.extensionConfig.ApplicationInsightsAnalytics.disableExceptionTracking)){n("_"+(r="onerror"));var o=a[r];a[r]=function(e,n,i,a,s){var c=o&&o(e,n,i,a,s);return!0!==c&&t["_"+r]({message:e,url:n,lineNumber:i,columnNumber:a,error:s}),c},e.autoExceptionInstrumented=!0}return t}(
    {
      instrumentationKey:"INSTRUMENTATION_KEY",
      endpointUrl: "https://dc.applicationinsights.azure.cn/v2/track"
    }
    );window[aiName]=aisdk,aisdk.queue&&0===aisdk.queue.length&&aisdk.trackPageView({});
</script>

```

**Comment by MariuszFerdyn** [comment]
Dobrze wiedzieć...

Pamiętajcie również, że AAD jest zupełnie niezależne w Azure China, czy też Germany. Co gorsza nie można również stosować kont gości. Czyli niby nie zintegrowane, ale jak w Azure China będziemy chcieli dodać konto Gościa z Azure Global to nam krzyknie, że nie jest to możliwe.

Dla międzynarodowych firm to niestety duże utrudnienie - zwłąszcza podczas adaopcji Office 365 i Teams.

**Reply by wi5nia** [reply]
Konto gościa można stosować jeśli też jest w danym cloud czytaj. China czy Germany ;)

**Reply by MariuszFerdyn** [reply]
No właśnie i dla mnie to słabe jest...

**Reply by wi5nia** [reply]
No wlasnie nie jest @MariuszFerdyn :) koncepcyjnie, jak i technologicznie "sovereign clouds" maja byc odseparowane calkowicie od reszty swiata inaczej by nie byly "sovereign" i byloby to bez sensu. Fakt, bywa to upierdliwe co zreszta teraz cwicze, ale zawsze mozna sobie zrobic federacje, PTA i inne zeby to uprosic.

**Reply by MariuszFerdyn** [reply]
No właśnie nie do końca spodziewanym zachowaniem byłoby aby po prostu zachowało się jako zupełnie zewnętrze konto jak @wp.pl dla którego zakłada się hasło i już.
Jeżeli chodzi o aplikacje to łatwo sobie poradzić za pomocą np. KeyCloak, w którym zrobiłem którym Cloudem chce się uwierzytelnić China czy Global, czy Germany.
Ba dorobiłem nawet, że można przejrzeć Teamsy w ten sposób - ale to cutsom aplikacja.

**Reply by wi5nia** [reply]
Ale nie mozesz zaprosic konta @wp.pl do AAD w Chinach bo jest odseparowane. IMHO to ma sens i tak bym sie spodziewal, ze tak dziala skoro jest odseparowane i ma byc odseparowane.

**Reply by kamil-mrzyglod** [reply]
Ja koncepcyjnie się zgadzam z @wi5nia - skoro mam China czy jakikolwiek inny Gov to z grubsza moim oczekiwaniem(jako klienta) byłoby całkowite zamknięcie na zewnętrzne tożsamości. 

Jaki jest tutaj business case - wybieram odseparowany region i nadal chcę wciągać konta z globala? To w końcu ten region jest odizolowany czy nie?

**Comment by RogalaPiotr** [comment]
Tak i nie. Jak masz firme ktora dziala na roznych rynkach to region China jest kolejnym z listy innych i jest to utrudnienie pod wieloma wzgledami. Jak masz jeszcze swoje DC to musisz poprawic swoj design o kolejne dostosowanie sie o region w China. Jak popatrzych na wytyczne to najlepiej odizolowac sie maksymalnie, ale zeby biznes dzialal dalej i wygodnie dla uzytkownika fajnie byloby uzywac tego samego “logina”

Pozdrawiam, Piotr Rogala
________________________________

**Reply by MariuszFerdyn** [reply]
Dokładnie...

Mam firmę globalną i powiedzmy oddział Nieniecki i oddział Chiński. Zgodnie z przepisami Chińskie dane muszą być w Chinach - ba nawet na VPNa trzeba mieć zgodę! I teraz jak chcesz założyć Team w Teamsach w Chinach - to musisz robić nowe konto dla tych z Niemiec. No a jak się korzysta z dwóch kont Teamsowych każdy o tym wie... (na gościu było by troszę kepiej)...
Nie mówiąc już o GAL etc.

Dla Goverment mogę to zrozumieć bardziej, zresztą tam obostrzenia dotyczące backgroud pracowników są itp./itd. To trochę inny temat wg. mnie i nie można tego łączyć.

**Reply by wi5nia** [reply]
Ale Chiny to mega specyficzny rynek gdzie nawet zeby domene dostac/hostowac musisz dostac zgode od urzedu wiec wcale mnie to nie dziwi.
Dwa - mylisz pojecia @RogalaPiotr - "fajnie byloby", a  "tak z zalozenia musi byc bo inaczej to nie ma sensu" - to dwie rozne rzeczy.

**Reply by kamil-mrzyglod** [reply]
No to jest przypadek, który rozumiem, ale tak czysto subiektywnie to jest koszt, który musisz ponieść w tym wypadku. Koszt zapewne akceptowalny po obu stronach(czyli i klienta i dostawcy).

**Reply by RogalaPiotr** [reply]
@wi5nia biznes ma to gdzies... z mojego punktu widzenia powinna zostac zachowana pelna izolacja i koniec, natomiast duze korpo unifikuja bo utrzymywanie dla nich 4 roznych azurów to jest problem... a do tego maja aws/gcp/ali... IT tutaj ma ulatwic a nie utrudnic ;)

**Reply by wi5nia** [reply]
To widac slaby biznes skoro nie rozumienia koncepcji sovereign cloud... skoro tego nie kuma to nie ma sensu sie pakowac w cos takiego jak Azure China - dostarczaj z public clouda bo mozesz w Chinach. Jak sie nagle Chinczycy wkurza i Cie zbalokuja to biznes szybciutko zrozumie czemu to tak jest.

**Reply by RogalaPiotr** [reply]
@wi5nia widze ze nic nie kumasz :D ale nie musisz sie zgadzac z calym swiatem bo i tak jest inaczej :)

**Reply by wi5nia** [reply]
To Ty nadal nie rozumiesz @RogalaPiotr ze to jest "by design" i tak musi byc bo inaczej to nie ma sensu

**Reply by MariuszFerdyn** [reply]
1. Dostarczanie spoza Chin jest wolne.
2. Trzymanie personal data poza Chinami jest nielegalne bez zgody.
3. Każda strona publiczna musi mieć numerek ICP
4. Każdy provider Chiński ułatwia zdobycie ICP

Więc z jednej strony jesteśmy skazani na dostawców Chińskich. Izolacja kont gości wg. mnie niczemu nie służy. Bo niby co izoluje?

**Reply by wi5nia** [reply]
Wiesz co @MariuszFerdyn to tez w druga strone mozna argument wytoczyc bo dzialy security bardzo czesto podnosza obawe przed faktem, ze bedzie takie konto tam (juz pomijam dokladnosc technikaliow), ale zdobeda twoj login/pass i zaczna przez to dostawac sie do innych systemow, w ktorych to konto jest (juz znowu dla uproszczenia pomijam MFA i inne) i juz izolacja gdzie masz konto tylko tam ma sens jako taki.

**Reply by RogalaPiotr** [reply]
@wi5nia chyba pomieszales wszystko razem, bo wyraznie napisalem ze z punktu widzenia korpo jest inaczej, a moj punkt widzenia jest spojny z Waszym czyli "izolacja". Skoro ja nie rozumiem to znaczy ze Ty chyba tez ;)

**Reply by wi5nia** [reply]
Nie zgodze sie, bo wiele korpo wlasnie zalezy na takiej izolacji w szczegolnosci na rynku/terenie takim jak Chiny - Germany pewnie byloby mniejszym problemem i cieszy sie wiekszym "zaufaniem", ale jednak Chiny sa na tyle specyficzne, ze ludzie staraja sie jak moga byc odlaczeni i biora na klate, ze tak musi byc.

**Reply by MariuszFerdyn** [reply]
Zamknijmy temat - bo i tak to konto zostanie tam założone i do utrzymania będą dwa konta z takimi sami hasłami pewnie, bo userowi łatwiej...

**Comment by wi5nia** [comment]
EOD
